test/tests/parallel/set1/spam_virus/fail2ban.bats

```diff
@@ -0,0 +1,203 @@
+    --env ENABLE_FAIL2BAN=1
+    --env POSTSCREEN_ACTION=ignore
+    --cap-add=NET_ADMIN
+    --ulimit "nofile=$(ulimit -Sn):$(ulimit -Hn)"
+  )
+  init_with_defaults
+  common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+  wait_for_smtp_port_in_container "${CONTAINER_NAME}"
+
+  local CONTAINER_NAME=${CONTAINER2_NAME}
+  local CUSTOM_SETUP_ARGUMENTS=(--env MAIL_FAIL2BAN_IP="$(docker inspect --format '{{ .NetworkSettings.IPAddress }}' ${CONTAINER1_NAME})")
+  init_with_defaults
+  common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
+}
+
+function teardown_file() {
+  docker rm -f "${CONTAINER1_NAME}" "${CONTAINER2_NAME}"
+}
+
+@test "${TEST_NAME_PREFIX} Fail2Ban is running" {
+  _run_in_container /bin/bash -c "ps aux --forest | grep -v grep | grep '/usr/bin/python3 /usr/bin/fail2ban-server'"
+  assert_success
+}
+
+@test "${TEST_NAME_PREFIX} localhost is not banned because ignored" {
+  _run_in_container /bin/sh -c "fail2ban-client status postfix-sasl | grep 'IP list:.*127.0.0.1'"
+  assert_failure
+  _run_in_container grep 'ignoreip = 127.0.0.1/8' /etc/fail2ban/jail.conf
+  assert_success
+}
+
+@test "${TEST_NAME_PREFIX} fail2ban-fail2ban.cf overrides" {
+  _run_in_container fail2ban-client get loglevel
+  assert_success
+  assert_output --partial 'DEBUG'
+}
+
+@test "${TEST_NAME_PREFIX} fail2ban-jail.cf overrides" {
+  for FILTER in 'dovecot' 'postfix' 'postfix-sasl'
+  do
+    _run_in_container /bin/sh -c "fail2ban-client get ${FILTER} bantime"
+    assert_output 1234
+
+    _run_in_container /bin/sh -c "fail2ban-client get ${FILTER} findtime"
+    assert_output 321
+
+    _run_in_container /bin/sh -c "fail2ban-client get ${FILTER} maxretry"
+    assert_output 2
+
+    _run_in_container /bin/sh -c "fail2ban-client -d | grep -F \"['set', 'dovecot', 'addaction', 'nftables-multiport']\""
+    assert_output "['set', 'dovecot', 'addaction', 'nftables-multiport']"
+
+    _run_in_container /bin/sh -c "fail2ban-client -d | grep -F \"['set', 'postfix', 'addaction', 'nftables-multiport']\""
+    assert_output "['set', 'postfix', 'addaction', 'nftables-multiport']"
+
+    _run_in_container /bin/sh -c "fail2ban-client -d | grep -F \"['set', 'postfix-sasl', 'addaction', 'nftables-multiport']\""
+    assert_output "['set', 'postfix-sasl', 'addaction', 'nftables-multiport']"
+  done
+}
+
+@test "${TEST_NAME_PREFIX} ban ip on multiple failed login" {
+  # can't pipe the file as usual due to postscreen
+  # respecting postscreen_greet_wait time and talking in turn):
+
+  # shellcheck disable=SC1004
+  for _ in {1,2}
+  do
+    docker exec "${CONTAINER2_NAME}" /bin/bash -c \
+    'exec 3<>/dev/tcp/${MAIL_FAIL2BAN_IP}/25 && \
+    while IFS= read -r cmd; do \
+      head -1 <&3; \
+      [[ ${cmd} == "EHLO"* ]] && sleep 6; \
+      echo ${cmd} >&3; \
+    done < "/tmp/docker-mailserver-test/auth/smtp-auth-login-wrong.txt"'
+  done
+
+  # need a bit more sleep as the test below that ´
+  # restarts F2B might interfere otherwise
+  sleep 30
```

Comment:
```suggestion
  # need a bit more sleep as the test below that (which restarts F2B) may interfere
  sleep 30
```

Which test is being referred to here? Ideally we can avoid a 30 second sleep..

Reply:
The current issue I faced with porting the F2B tests is that the test that restarts F2B can interfere with other tests apparently, when they are executed at the sime time. I was under the impression BATS would run two different test _files_ at the same time, but from what I was seeing yesterday, `@test`s were run in parallel too.

Uhhh, ohhh; I just wrote the above text, searched for it and found the solution: `--no-parallelize-within-files`. I will provide an update for this.

Reply:
Done in 0902ab2f4ec2bb73350039702ba7b6cd111a9ac3

Reply:
> Uhhh, ohhh; I just wrote the above text, searched for it and found the solution: `--no-parallelize-within-files`. I will provide an update for this.

I haven't looked at the related docs in some time, but I think there may have been a way to selectively indicate to BATS which files should opt-out of this kind of parallelization.

For now though without looking through each files test cases in sequence, it's probably better to opt-out completely like you've gone ahead and done :+1: 

I know with a test like the cipherlists where each test case takes a fair chunk of time on it's own (_in total I think it's about 25-33% of the serial test time due to some notable slowdown overhead caused by dockers networking_), it'd be useful to run each test case in parallel. But with your efforts here, it looks like we could also split out common logic into a helper and have each test case as it's own file grouped in a folder ;)

---

Feel free to resolve this once you've read the message :+1: